src/index.js

```diff
@@ -1,62 +1,170 @@
 
-import hash from 'murmurhash-js/murmurhash3_gc'
-import createRules from './create-rules'
 import sheet from './sheet'
+import hash from './hash'
+import {
+  isArr,
+  isObj,
+  parseValue,
+  clean,
+  kebab,
+  dot,
+  flattenArray,
+  flattenArrayValues,
+} from './util'
+import shorthands from './shorthands'
 
 export const cache = []
 
-const cxs = (...args) => {
-  const selector = typeof args[0] === 'string' ? args[0] : null
-  const style = args.reduce((a, b) => {
-    return a || typeof b === 'object' ? b : null
-  }, null)
-  const classNames = []
-  const hashname = 'cxs-' + hash(JSON.stringify(style), 128)
-  const rules = createRules(selector || hashname, style)
-
-  rules.filter(r => !(/:/.test(r.selector)))
-    .filter(r => !(/\s/.test(r.selector)))
-    .forEach(r => {
-      classNames.push(r.selector.replace(/^\./, ''))
-    })
-
-  rules
-    .filter(r => cache.indexOf(r.id) < 0)
-    .forEach(r => {
-      cache.push(r.id)
-      sheet.insert(r.css)
-    })
-
-  return classNames.reduce((a, b) => {
-    if (a.indexOf(b) > -1) return a
+// Insert rule into stylesheet and cache
+const insert = rule => {
+  if (cache.indexOf(rule.className) > -1) {
```

Comment:
Why use array instead of hash? This seems like it'd be very wasteful, especially for larger applications with hundreds of classes.